app/src/main/java/ch/epfl/sdp/blindly/recording/AudioLibraryAdapter.kt

```diff
@@ -0,0 +1,218 @@
+        val remainingTimer: Chronometer = view.findViewById(R.id.remainingTimer)
+        val selectButton: Button = view.findViewById(R.id.selectButton)
+
+        init {
+            // Define click listener for the ViewHolder's View.
+            nameDurationLayout.setOnClickListener(this)
+            expandableLayout.visibility = View.GONE
+        }
+
+        override fun onClick(v: View?) {
+            // Forward the click to the main activity
+            val position = adapterPosition
+            if (position != RecyclerView.NO_POSITION) {
+                listener.onItemClick(position)
+            }
+        }
+    }
+
+    // Expands or collapse the layout for playing the audio file
+    private fun toggleLayout(isExpanded: Boolean, v: View, layoutExpand: RelativeLayout): Boolean {
+        if (isExpanded) {
+            RecordAnimations.expand(layoutExpand)
+        } else {
+            RecordAnimations.collapse(layoutExpand)
+        }
+        return isExpanded
+    }
+
+    // Create new views (invoked by the layout manager)
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        // Create a new view, which defines the UI of the list item
+        val view = LayoutInflater.from(context).inflate(R.layout.audio_recording, null)
+        return ViewHolder(view)
+    }
+
+    // Replace the contents of a view (invoked by the layout manager)
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        // Get element from your dataset at this position and replace the
+        // contents of the view with that element
+        viewHolder.recordName.text = recordList[position].name
+        viewHolder.recordDuration.text = recordList[position].durationText
+        viewHolder.nameDurationLayout.setOnClickListener {
+            val show: Boolean = toggleLayout(!recordList[position].isExpanded, it, viewHolder.expandableLayout)
+            recordList[position].isExpanded = show
+        }
+
+        val playBar = viewHolder.playBar
+        val playTimer = viewHolder.playTimer
+        val remainingTimer = viewHolder.remainingTimer
+        val playPauseButton = viewHolder.playPauseButton
+        val movePlayBarThread = createPlayBarThread(playBar)
+
+        setCountDownTimer(remainingTimer)
+
+        viewHolder.playPauseButton.setOnClickListener {
+            if (isPlayerStopped) {
+                createMediaPlayer(recordList[position].filePath)
+                remainingTimer.base = SystemClock.elapsedRealtime() + mediaPlayer!!.duration.toLong()
+
+                mediaPlayer?.setOnCompletionListener {
+                    mediaPlayer?.stop()
+
+                    setCompletedAudioView(playTimer, remainingTimer, playPauseButton)
+                }
+            }
+            if (!mediaPlayer!!.isPlaying) {
+                mediaPlayer?.start()
+
+                if (!isPlayerPaused) {
```

Comment:
I'm not sure I understand that block of code. Line 112-113 says "if the mediaPlayer isn't playing, you start it", so it's now playing, right ? So what's the use of the condition here in line 115? Isn't the player always playing after you start it ?

Reply:
As you saw in my code, the booleans are modified in the functions `setStateView`. Then, even if I called `mediaPlayer.start()`, the boolean hasn't been modified there. 
If the player isn't paused, the timer resets. Otherwise, it keeps its current value.
Is it clearer now ?